pandas/core/generic.py

```diff
@@ -4136,9 +4136,16 @@ def fillna(self, value=None, method=None, axis=None, inplace=False,
                 return result if not inplace else None
 
             elif not is_list_like(value):
-                new_data = self._data.fillna(value=value, limit=limit,
+                if axis == 0:
+                    new_data = self._data.fillna(value=value, limit=limit,
```

Comment:
this is very verbose. further you could be downcasting twice.

Reply:
Is this block any better?

```python
            elif not is_list_like(value):
                if axis == 0:
                    target = self._data
                elif axis == 1:
                    target = self.T
                new_data = target.fillna(value=value, limit=limit,
                                         inplace=inplace,
                                         downcast=downcast)
                if axis == 1:
                    new_data = new_data.T
```